test/models/auto_suggestion_test.rb

```diff
@@ -0,0 +1,50 @@
+require 'test_helper'
+
+class AutoSuggestionTest < ActiveSupport::TestCase
+
+  def test_existence
+    assert GeneticAlgorithm
+  end
+
+  def test_auto_suggestion
+    generate_devs(150)
+
+    contractor_opts = {
+      "front_end" => [3,"Javascript"],
+      "back_end" => [1,"Ruby"],
+      "database" => [3,"Postgresql"],
+      cost: 150,
+      total_devs: 7
+    }
+
+    ga = GeneticAlgorithm.new(dev_opts(contractor_opts))
+    init = ga.fittest_chromosome.dna
+    p Fitness.developers_fitness(contractor_opts)[init]
+    p init.map{|x| {x.specialty.name => x.tags.map{|y| y.name}}}
```

Comment:
Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.<br>Space inside { missing.<br>Space inside } missing.<br>Pass `&:name` as an argument to `map` instead of a block.